src/python/example_tensor.py

```diff
@@ -0,0 +1,325 @@
+
+  def __init__(self, shape=None, device=None, dtype=kFloat32):
+
+    self.tuple_shape = shape
+    self.device = device
+    self.dtype = dtype
+
+    if shape == None:
+      self.singa_tensor = singa.Tensor()
+    else:
+      vs = tuple_to_vector(shape)
+      if device == None:
+        self.singa_tensor = singa.Tensor(vs, dtype)
+      else:
+        self.singa_tensor = singa.Tensor(vs, device, dtype)
+
+  def copy_Tensor(self, t):
+    self.tuple_shape = t.tuple_shape
+    self.device = t.device
+    self.dtype = t.dtype
+    self.singa_tensor = singa.Tensor(t.singa_tensor)
+
+  def data(self):
+    idx = self.singa_tensor.data_type()
+    if idx == kFloat32:
+      data_array = singa.floatArray_frompointer(self.singa_tensor.floatData())
+      dt = np.float32
+    elif idx == kFloat16:
+      print 'not implemented yet'
+      return
+      #data_array = singa.floatArray_frompointer(self.singa_tensor.floatData())
+      #dt = np.float16
+    elif idx == kInt:
+      data_array = singa.intArray_frompointer(self.singa_tensor.intData())
+      dt = np.int32
+    elif idx == kChar:
+      data_array = singa.charArray_frompointer(self.singa_tensor.charData())
+      dt = np.int8
+    elif idx == kDouble:
+      data_array = singa.doubleArray_frompointer(self.singa_tensor.doubleData())
+      dt = np.float64
+
+    data = [data_array[i] for i in range(self.singa_tensor.Size())]
+    data = np.array(data, dtype=dt).reshape(self.tuple_shape)
+    return data
+
+  def data_type(self):
+    return self.singa_tensor.data_type()
+
+  def shape(self, axis=None):
+    if axis==None:
+       return self.singa_tensor.shape()
+    else:
+       return self.singa_tensor.shape(axis)
+
+  def ndim(self):
+    return self.singa_tensor.nDim()
+
+  def transpose(self):
+    return self.singa_tensor.transpose()
+
+  def size(self):
+    return self.singa_tensor.Size()
+
+  def memsize(self):
+    return self.singa_tensor.MemSize()
+
+  def reshape(self, shape):
+    self.tuple_shape = shape
+    self.singa_tensor.Reshape(tuple_to_vector(shape))
+
+  def reset_like(self, t):
+    self.singa_tensor.ResetLike(t.singa_tensor)
+
+  def as_type(self, dtype):
+    self.singa_tensor.AsType(dtype)
+
+  def to_device(self, device):
+    self.singa_tensor.ToDevice(device)
+
+  def to_host(self):
+    self.singa_tensor.ToHost()
+
+
+  def set_value(self, x):
+    self.singa_tensor.SetValue(x)
+
+  def copy_data_from_hostptr(self, ptr, num):
+    self.singa_tensor.CopyDataFromHostPtr(ptr, num)
+
+  def copy_data(self, t):
+    self.singa_tensor.CopyData(t.singa_tensor)
+
+  def clone(self):
+    new_t = Tensor(self.tuple_shape, self.device, self.dtype)
+    new_t.singa_tensor = self.singa_tensor.Clone()
+    return new_t
+
+  def matrix_transpose(self):
+    new_t = Tensor(self.tuple_shape, self.device, self.dtype)
+    new_t.singa_tensor = self.singa_tensor.T()
+    return new_t
+
+  # TODO(chonho-02) assign
+  # same as copy_Tensor(t) ???
+  def assign(self, t):
```

Comment:
in c++, the copy constructor and assgin operator would actually not copy the data values, instead they just share the input tensor's data pointer (i.e, the blob).
for python, we need to check whether there is a operator=() function and copy constructor. If it has then we should override them like C++.